lib/factory_bot/trait.rb

```diff
@@ -7,9 +7,12 @@ def initialize(name, &block)
       @name = name.to_s
       @block = block
       @definition = Definition.new(@name)
-
       proxy = FactoryBot::DefinitionProxy.new(@definition)
-      proxy.instance_eval(&@block) if block_given?
+
+      if block_given?
+        evaulated_block = proxy.instance_eval(&@block)
+        ensure_trait_not_self_referencing(evaulated_block)
```

Comment:
It seems exactly what it needs, I may be completely out of sync here, but this seems possible:
```ruby
  factory :user do
    trait :admin do
      admin {true}
    end
end
```
And shouldn't that syntax be valid?

Reply:
It seems from the issue https://github.com/thoughtbot/factory_bot/issues/1238 that when you have a self-referencing trait, it will call itself recursively.